server/src/main/java/org/opensearch/cluster/health/ClusterShardHealth.java

```diff
@@ -113,7 +114,7 @@ public final class ClusterShardHealth implements Writeable, ToXContentFragment {
     private int delayedUnassignedShards;
     private final boolean primaryActive;
 
-    public ClusterShardHealth(final int shardId, final IndexShardRoutingTable shardRoutingTable) {
+    public ClusterShardHealth(final int shardId, final IndexShardRoutingTable shardRoutingTable, final IndexMetadata indexMetadata) {
```

Comment:
Similar to comment on getShardHealth, If all we need to know is if the cluster block is applied.  Lets overload the ctor here with a boolean and default it to false?

Reply:
I have updated this in my latest commit, please check.